### PR TITLE
Setup redux

### PR DESCRIPTION
**Pull request by Gopxfs**
# :ledger: Summary
Added redux setup to the project, including reducers, actions, and store.

### books/books.js :link:
Contains actions to add and remove a book, using the booksReducer.
### categories/categories.js :link:
Contains an action to check the status, using the categoriesReducer.
### configureStore.js
Merge together the reducers and export the store, containing a rootReducer.

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (keneogu) [inline src/redux/categories/categories.js]: This should follow the [duck pattern](https://github.com/erikras/ducks-modular-redux) e.g `const CHECK_STATUS = bookstore/categories/CHECK_STATUS`

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (keneogu) [inline src/redux/books/books.js]: This should follow the [duck pattern](https://github.com/erikras/ducks-modular-redux) e.g `const CHECK_STATUS = bookstore/books/CHECK_STATUS`

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  switch (action) {
+    case ADD_BOOK: {
+      return {
+        ADD_BOOK: true,
+      };
+    }
+    case REMOVE_BOOK: {
+      return {
+        REMOVE_BOOK: true,
+      };
```
Comment (keneogu) [inline src/redux/books/books.js]: This should return a new array, which will not contain the book you want to remove from the store (filter by the id key - i.e.: return state.filter(book => book.id !== id); The function should use es6 filter method.

**Review by keneogu — CHANGES_REQUESTED** [review]
Hi @Gopxfs,

Good job so far!

- All linters are passing
- You are following GitHub flow
- Good commit history

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @Gopxfs, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**